### ISSUE TITLE
[Feature Request] Sponsorblock integration

### ISSUE THREAD

**Issue by DraconicNEO** [yuliskov/SmartTubeNext#11]
I think it would be a good idea to have Sponsorblock integration added to SmartTubeNext because it would improve the user experience by blocking ads that can't be blocked by adblockers. Other YouTube clients like YouTube Vanced and NewPipe also have Sponsorblock support. Unfortunately there aren't any Android TV YouTube clients with Sponsorblock support which is a shame because unlike on mobile skipping portions of videos on Android TV is tedious due to the limited controls.
Bellow are some links to some information on Sponsorblock that might be useful.

Sponsorblock Resources

[Sponsorblock Source](https://github.com/ajayyy/SponsorBlock)

[YTVanced Version](https://github.com/YTVanced/SponsorBlock)

**Comment by AtlAntA118** [comment]
Would like to second this request. Been using it on my phone with vanced and its really helpfull and usefull.

**Comment by hussainalhaddad** [comment]
This would be a dream come true!!

**Comment by darkangelus** [comment]
I too would appreciate this feature.

**Comment by steve51184** [comment]
The ad blocking is working perfectly for me so adding sponsor block would be the icing on the cake

**Comment by Vkuparin** [comment]
Logged in just to voice my support for this! Sponsorblock is amazing.

**Comment by fawzib** [comment]
i also vote for this. amazing and a must have feature. might get into the code and see if I can implement it.

**Comment by rix1337** [comment]
Is there a bounty program for this app?
Clearly people want this feature, but cannot provide code themselves.

**Comment by pekempy** [comment]
Agreed. It doesn't even need the ability to add segments, just grab the timestamps from sponsorblock and skip particular sections

**Comment by ChaosBlades** [comment]
This app is the only place I watch YouTube that doesn't have SponsorBlock. Sometimes on videos I know that have a lot of sponsor segments I just use my phone instead of SmartTube Next.

I agree it will not need the options to submit segments. It will need the settings menu to select if you want to auto skip or manually skip the different types of segments. It will also need an option to turn off SponsorBlock on a per video basis in case of false positives/segments reported improperly.

**Comment by rix1337** [comment]
All features you listed are optional for a working first implementation.

This request should first and foremost be about enabling sponsor block at all.

**Comment by superxtc** [comment]
Sponsorblock would be amazing.

**Comment by ManuLinares** [comment]
Sponsorblock should make this app essential for any Android TV

**Comment by DraconicNEO** [comment]
Wow This request blew up very fast. I hope this actually gets accepted

**Comment by ArestouMars** [comment]
I agree with this. YouTube is so crappy right now that they don't see a problem that people are advertising on their videos outside of YouTube where they get all the cut.

If there is a reason not to add it for ethics, a small amount of people using this app won't affect creators as I feel like only people who aren't terrible with technology are able to install and sideload apps like this.

**Comment by DraconicNEO** [comment]
> 
> 
> I agree with this. YouTube is so crappy right now that they don't see a problem that people are advertising on their videos outside of YouTube where they get all the cut.
> 
> If there is a reason not to add it for ethics, a small amount of people using this app won't affect creators as I feel like only people who aren't terrible with technology are able to install and sideload apps like this.

If the dev had any ethical issues towards ad blocking this project just straight up wouldn't exist, as this is basically its main function.

**Comment by rodalpho** [comment]
Well no, it's also used on a lot of chinese android set top boxes that don't support android TV. But I agree the developer clearly isn't _opposed_ to blocking ads.

**Comment by f-o** [comment]
Here's to hoping, though I'm afraid it might not be implemented.

**Comment by fawzib** [comment]
wonder why the dev never commented on this ever. a road map would actually be great

**Comment by KruyKaze** [comment]
Dev said in telegram chat that it's planned for next year. Hoping for
January ☺

**Comment by rodalpho** [comment]
Fantastic news!!

**Comment by DraconicNEO** [comment]
> Fantastic news!!

It sure is 🙂

**Comment by akuiraz86** [comment]
+1
SponsorBlock would be a great addition!

**Comment by IPv777** [comment]
+1

**Comment by rix1337** [comment]
Latest release contains the Sponsors Block content blocking setting and it works. 

Thank you! Issue closed?

**Comment by rix1337** [comment]
Just tested the implementation in the latest release.

This has been solved!!! Thanks 😊

**Comment by rodalpho** [comment]
Yes, it works great! Thank you @yuliskov!!

**Comment by foreveryoungforty** [comment]
@DraconicNEO implemented in new update - please close this issue.

**Comment by ChaosBlades** [comment]
The issue is not resolved. 
1. It is only blocking sponsors segments and not the other five types you can block from what I can see. Like intros.
2. It isnt highlighting the section's on the progress bar.

**Comment by rix1337** [comment]
Those are additional requests not specified in the original post.

Since they are valid, open other issues for both?

**Comment by fawzib** [comment]
true, I know the author will improve and add more features so no worries

**Comment by rix1337** [comment]
It just best to not change the scope of an issue after its original request has been resolved.

Further options should thus be treated as differend issues.

**Comment by fawzib** [comment]
I opened a new ticket
https://github.com/yuliskov/SmartTubeNext/issues/385

**Comment by DraconicNEO** [comment]
I'm Closing this as the Issue has been Resolved
